pkg/vault/test/authorizer/TimelockAuthorizerExecute.test.ts

```diff
@@ -0,0 +1,431 @@
+    });
+
+    it('reverts if the target is the authorizer', async () => {
+      const where = authorizer.instance;
+      await expect(schedule(where)).to.be.revertedWith('CANNOT_SCHEDULE_AUTHORIZER_ACTIONS');
+    });
+
+    it('reverts the target is the execution helper', async () => {
+      const where = await authorizer.instance.getTimelockExecutionHelper();
+      await expect(schedule(where)).to.be.revertedWith('ATTEMPTING_EXECUTION_HELPER_REENTRANCY');
+    });
+  });
+
+  describe('execute', () => {
+    const delay = DAY;
+    const functionData = '0x0123456789abcdef';
+
+    sharedBeforeEach('grant protected function permission with delay', async () => {
+      // We must set a delay for the `setAuthorizer` function as well to be able to give one to `protectedFunction`
+      const setAuthorizerAction = await actionId(vault, 'setAuthorizer');
+      await authorizer.scheduleAndExecuteDelayChange(setAuthorizerAction, delay, { from: root });
+
+      const protectedFunctionAction = await actionId(authenticatedContract, 'protectedFunction');
+      await authorizer.scheduleAndExecuteDelayChange(protectedFunctionAction, delay, { from: root });
+      await authorizer.grantPermission(protectedFunctionAction, user, authenticatedContract, { from: root });
+    });
+
+    const schedule = async (executors: SignerWithAddress[] | undefined = undefined): Promise<number> => {
+      const data = authenticatedContract.interface.encodeFunctionData('protectedFunction', [functionData]);
+      return authorizer.schedule(authenticatedContract, data, executors || [], { from: user });
+    };
+
+    it('can execute an action', async () => {
+      const id = await schedule();
+      await advanceTime(delay);
+      const receipt = await authorizer.execute(id, { from: executor });
+
+      expectEvent.inIndirectReceipt(await receipt.wait(), authenticatedContract.interface, 'ProtectedFunctionCalled', {
+        data: functionData,
+      });
+    });
+
+    it('action is marked as executed', async () => {
+      const id = await schedule();
+      await advanceTime(delay);
+      await authorizer.execute(id, { from: executor });
+
+      const scheduledExecution = await authorizer.getScheduledExecution(id);
+      expect(scheduledExecution.executed).to.be.true;
+    });
+
+    context('when the action is protected', () => {
+      it('all executors can execute', async () => {
+        const id = await schedule([executor, account]);
+        await advanceTime(delay);
+
+        expect(await authorizer.isExecutor(id, executor)).to.be.true;
+        expect(await authorizer.isExecutor(id, account)).to.be.true;
+
+        const receipt = await authorizer.execute(id, { from: account });
+        expectEvent.inIndirectReceipt(
+          await receipt.wait(),
+          authenticatedContract.interface,
+          'ProtectedFunctionCalled',
+          {
+            data: functionData,
+          }
+        );
```

Comment:
If you really want to, you can use snapshots to undo the execute and then re-do it from the other account. But tbh I think that's overkill.